utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/jsonpayload/JsonPayloadToAvroConverter.scala

```diff
@@ -0,0 +1,146 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.jsonpayload
+
+import io.circe.Decoder
+import io.circe.Json.fromString
+import org.apache.avro.Conversions.UUIDConversion
+import org.apache.avro.data.TimeConversions
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.specific.SpecificRecordBase
+import org.apache.avro.{LogicalTypes, Schema}
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.jsonpayload.JsonPayloadToAvroConverter._
+import tech.allegro.schema.json2avro.converter.types.AvroTypeConverter
+import tech.allegro.schema.json2avro.converter.{CompositeJsonToAvroReader, JsonAvroConverter}
+
+import java.math.RoundingMode
+import java.time.{Instant, LocalDate, LocalTime}
+import java.util
+
+class JsonPayloadToAvroConverter(specificClass: Option[Class[SpecificRecordBase]]) {
+
+  private val converter = new JsonAvroConverter(new CompositeJsonToAvroReader(
+    DateConverter, TimeMillisConverter, TimeMicrosConverter, TimestampMillisConverter, TimestampMicrosConverter,
+    UUIDConverter, DecimalConverter
+  ))
+
+  def convert(payload: Array[Byte], schema: Schema): GenericRecord = {
+    specificClass match {
+      case Some(kl) => converter.convertToSpecificRecord(payload, kl, schema)
+      case None => converter.convertToGenericDataRecord(payload, schema)
+    }
+  }
+
+}
+
+object JsonPayloadToAvroConverter {
+
+  object DateConverter extends AvroTypeConverter {
+    private val conversion = new TimeConversions.DateConversion
+
+    override def canManage(schema: Schema, path: util.Deque[String]): Boolean =
+      schema.getType == Schema.Type.INT && schema.getLogicalType == LogicalTypes.date()
+
+    override def convert(field: Schema.Field, schema: Schema, jsonValue: AnyRef, path: util.Deque[String], silently: Boolean): AnyRef = {
+      jsonValue match {
+        case number: Number => conversion.fromInt(number.intValue(), schema, schema.getLogicalType)
+        case str: String => Decoder[LocalDate].decodeJson(fromString(str)).fold(throw _, identity)
+        case other => other
```

Comment:
do you think it's sensible to pass anything here? Wouldn't it be better to throw exception for sth unexpected?

Reply:
I wrongly understand concept of fallbacking to other options in this converter. it looks that I should handle `silently` parameter and return INCOMPATIBLE or throw exception. Changed and added test for union showing where is a problem.